### PR TITLE
Fix provider runtime mounts and context links

### PR DESCRIPTION
**Pull request by kaspernj**
## Summary
- add durable canonical Codex/Kimi/OpenCode runtime and immutable agent-context mounts to the development Compose lane
- preserve and validate canonical runtime aliases while keeping OpenCode session/state directories lane-local
- reject redirected alias parents and Docker bind auto-creation
- wire container-native Threadwire provider executables
- add focused bootstrap tests, static negative probes, documentation, and a changelog fragment

## Validation
- provider bootstrap scenarios passed, including redirect-chain regressions
- `npm run verify:docker-dev-environment` passed
- `npm run lint` (ESLint, typecheck, Fallow) passed
- Bash syntax, ShellCheck, Compose render, and `git diff --check` passed
- real tracked-only Compose recreation passed
- Codex and OpenCode fresh + exact resume passed through Threadwire
- Kimi reached the real provider; account currently returns its billing-cycle quota 403
- independent review and terminal re-review passed

### CODE REVIEW

File: docs/docker-development-environment.md [hunk 1]
```diff
@@ -6,14 +6,64 @@ The checked-in root `Dockerfile` and `compose.yml` define one canonical `dev` se
...
+test -d "$github_config_source"
+test ! -L "$AGENT_CONTEXT_SOURCE_PATH"
+test -f "$agent_context_source/AGENTS.md" && test -d "$agent_context_source/skills"
+test "$(basename -- "$AGENT_CONTEXT_SOURCE_PATH")" != current
+
+case "$provider_runtime_source/" in "$dev_home_source/"*) exit 1;; esac
+case "$dev_home_source/" in "$provider_runtime_source/"*) exit 1;; esac
+case "$agent_context_source/" in "$dev_home_source/"*) exit 1;; esac
+case "$dev_home_source/" in "$agent_context_source/"*) exit 1;; esac
+case "$agent_context_source" in /opt/data|/opt/hermes-dind-shared|/opt/hermes-dind-shared/agent-context) exit 1;; esac
```
[the remark this answers is not in the document]
Comment (kaspernj) [reply docs/docker-development-environment.md]: Addressed in 3bba4b68. The documented host preflight now rejects bidirectional resolved containment between provider-runtime and agent-context sources, and the checked-in verifier plus negative probes enforce both checks. Checks: focused bootstrap test, verify:docker-dev-environment, Bash syntax, ShellCheck, npm lint/typecheck/Fallow, YAML parse, and git diff --check passed.

File: scripts/bootstrap-provider-runtime.sh [hunk 2]
```diff
@@ -0,0 +1,226 @@
...
+
+  if [[ -L $target_path ]] && [[ $(readlink -- "$target_path") == "$source_path" ]]; then
+    return
+  fi
+
+  if [[ -e $target_path || -L $target_path ]]; then
+    preserve_conflict "$target_path"
+  fi
+
+  if ! ln -s -- "$source_path" "$target_path"; then
```
[the remark this answers is not in the document]
Comment (kaspernj) [reply scripts/bootstrap-provider-runtime.sh]: Addressed in 3bba4b68. Shared Codex/Kimi discovery-link validation and migration now take an exclusive flock on the provider-runtime directory, revalidate aliases after lock acquisition, and release the lock before lane-local setup. A deterministic two-lane regression proves both starts succeed, aliases remain exact, and the shared conflict is backed up once. All focused and aggregate gates passed.
